app/services/reporter.rb

```diff
@@ -6,38 +6,76 @@
 # run queries and produce reports from the results, for consumption
 # by preservation catalog maintainers
 class Reporter
-  # @param [String] the name of the storage root for which druids should be listed
+  attr_reader :storage_root, :druids
+
+  # @param [String] the name of the storage root to initialize for reporter
+  # @return [Reporter] the reporter
+  def initialize(params)
+    @storage_root = MoabStorageRoot.find_by!(name: params[:storage_root_name])
+    @druids = []
+    moab_storage_root_druid_list
+  end
+
+  # @return [Array] an array of druids on the storage root
+  def moab_storage_root_druid_list
+    PreservedObject
+      .joins(:complete_moabs)
+      .where(complete_moabs: { moab_storage_root: storage_root })
+      .select(:druid, :status)
+      .order(:druid)
+      .each_row do |po_hash|
+        @druids << po_hash['druid']
+      end
+  end
+
+  # @param [Array] druids to output details for
+  # @param [Boolean] optionally only output lines with audit errors
+  # @return [Array] an array of hashes with details for each druid provided
+  def moab_detail_for(data, errors_only: false)
```

Comment:
Open to renaming `data` to something more intention-revealing, like `druid_list` or `druids`?